### DISCUSSION TITLE
Some roles from OpenCorporates - what can/should be added to FIBO?

### DISCUSSION THREAD

**Discussion by trypuz** [edmcouncil/fibo#1359] [Business Entities (BE)]
1. accountant
2. accounts manager
3. adm. dir
4. advisor
5. agent
6. associate director
7. board manager
8. board of advisors
9. book keeper
10. businessman
11. coo
12. head of products
13. chairman
14. chairperson
15. chartered accountant
16. chief creative officer
17. civil engineer
18. co director
19. commodity trader
20. company chairman
21. company director
22. company formation agent
23. company secretary
24. consultant
25. control board member
26. control board support
27. general pm
28. deputy manager
29. director
30. director of communications
31. director of hr services
32. director of hr shared services
33. entrepreneur
34. executive chairman
35. executive director
36. finance director
37. financial accountant
38. financial adviser
39. general manager
40. head of pm department
41. initial board of director
42. internet professional
43. investment banker
44. investment executive
45. investor
46. manager
47. managing director
48. merger & acquisition manager
49. mortgage adviser
50. mortgage consultant
51. non-executive director
52. president
53. private equity investor
54. private equity investor execut
55. project manager
56. sales director
57. secretary
58. sole trader
59. solicitor
60. top executive
61. treasurer
62. vice president

**Comment by ElisaKendall** [comment]
@trypuz Hi Robert -- we actually have a few of these that you might want to map to what you found.  They include:
1. For account manager, in the US that role is called relationship manager -- we have that, but maybe we need a synonym?
2. agent - we have 3rd party agent, legal agent, licensed agent, registered agent, agent for service of process - so it would depend on the definition of agent
3. advisor is on my list, but for financial advisor only ***
4. board manager - is this the same as the chairperson of a board of directors?  If so, we can add that one ***, which would be the same as chairman, chairperson, company chairman, company director - we do have board member, which you could map to, and we have inside/outside board member as well if you know which applies
5. is a board of advisors distinct from a board of directors?  If so, we will have to think about whether or not it is within scope
6. COO - we could add them as needed, but we do have corporate officer you could map to for now
7. Chartered accountant - we have auditor, but that's different from an accountant for a firm, and the question would be do we need the accounting firm and/or the individual person that is the responsible party / primary partner from an accountancy that is responsible for signing off on the work that the outside accounting team performed?
8. chief creative officer - map to corporate officer
9. company formation agent - this is the same as an agent for service of process, which we do have
10. company secretary - this is in BE executives
11. director - if they are a board member then we would map to that.  if they are an employee with the title of director then no
12. executive chairman - this is the same as chief executive officer
13. executive director - this depends - we likely don't have this, which would be a corporate officer for a not for profit, but you could use corporate officer
14. financial advisor might be in Dev - we should think about adding this ...
15. investment banker - we have investment bank, but not the role for the person, but you could relate them to the organization using organization member
16. initial board of directors - ?? we have board of directors
17. we have Investor
18. managing director - this is on my list ***
19. mortgage advisor - if this is not in loans we should add it ***
20. non-executive director -- this is the same as non-executive board member
21. president - we should add this ***
22. private equity investor -- maps to investor
23. secretary -- we have this if it is company/corporate secretary
24.  sole trader - this is there, maps to sole proprietor
25. top executive -- maps to chief executive officer
26. treasurer -- might be the same as chief financial officer ?  we have that

So - there are a few gaps I would add, but a number of the others are out of scope for FIBO I think, including the ones marked ***
I'm also considering adding an employment situation, with employer and employee roles.  Some of the other roles could be mapped to employee ... most banks have a gazillion vice presidents ...

**Comment by rivettp** [comment]
The broader question is HOW should these be represented? To answer that @trypuz we need some use cases, which I know is something you champion: are there some for KYC?

Without any use cases, it would seem to me that none of these have any distinct properties or semantics, so they could each be represented as individuals of (new) class JobTitle ("Role" is overused), which would be referenced from individuals of class Position (see my response to #1236).

**Comment by rivettp** [comment]
What we do have in FIBO already is probably over-complex for the majority of use cases I can conceive of for use of Open Corporates data. It really needs testing with some example data. As a mind experiment I tried jotting down what might be needed to represent the CompanySecretary of a Company and gave up after 18 classes. It seems specific individuals would be needed for leaf classes PubliclyHeldCompany, NaturalPerson, CompanySecretary, Authorization, AuthorizingParty, ControlledParty. And it gets very confusing as to whether the Company is the primary Actor because it is the AuthorizingParty or the Secretary is in control as a CorporateOfficer.

**Comment by ElisaKendall** [comment]
AML, and KYC for that matter, require linking who knows who through the networks they participate in and their roles in those networks.  Use of a property that links people to a generic JobTitle doesn't do that.  What's important about the representation we currently have is being able to create that linkage.  Obviously something is amiss if you had this much difficulty in doing so.  perhaps Dean can provide some insights, since he was able to successfully use these patterns for a client.  I've done the same, but not at the same scale.

**Comment by rivettp** [comment]
Linking Position (not Person directly) to JobTitle was in addition to my proposal as comment on #1236 that I referenced. That gives you the people associations needed: Person to Position to Org to other Positions and other People in the same Org. JobTitle would be incidental to that Person-Person linkage if further detail is needed. Importantly it does allow that person interconnectivity to be established without worrying about different specific subclasses, roles and situations. The current verbose FIBO model doesn't do that any better.

It may be that @dallemang was able to use current FIBO more efficiently for people-people connections, but, as you say, "something is amiss" if FIBO is lacking use cases, documentation or even examples in this application area such that even different experts end up with quite different solutions (I know FIBO as well as most). One long-standing issue is that FIBO, and OWL more generally, has no way of documenting usage patterns for the number of individuals used to reify the different classes, including Roles.

**Comment by jfgemski** [comment]
Here’s my thoughts on this.

1.	“role” vs “position” - The broader term is “role” such as the role of a department within an organization
2.	Role has different enumerations based on its context

*	As given above, the role of a department has different values than the role of an employee

3.	I wouldn’t use “position” in the context of a person’s role in an organization within FIBO since “position” in the investment industry refers to “Investor’s stake in a particular security or market”. (Barron’s)
4.	Even the role of an employee in an organization depends on the type of organization.   For example, for someone working at an exchange the roles would be trader, market-maker, specialist, etc.
5.	Please don’t link role (or position) to title.  For example, my title was Vice President  of Product Management while my role was manager of the data model product.  I’ve found that titles rarely specifies what the role actually is.

*	Also, please don’t even try to enumerate titles since that’s more of an HR responsibility and varies widely among organizations.  A person given a title of VP in one organization will likely have way different roles and responsibilities from a VP in another organization.

6.	Please don’t even try to enumerate roles as what Robert’s email seems to want to do.  The problem is as I’ve been giving above, a role is highly dependent on the context.

John

 

Jfgemski@gmail.com <mailto:Jfgemski@gmail.com>  | M: (516) 353-0004

**Comment by trypuz** [comment]
> @trypuz Hi Robert -- we actually have a few of these that you might want to map to what you found. They include:
> 
> 1. For account manager, in the US that role is called relationship manager -- we have that, but maybe we need a synonym?
> 2. agent - we have 3rd party agent, legal agent, licensed agent, registered agent, agent for service of process - so it would depend on the definition of agent
> 3. advisor is on my list, but for financial advisor only ***
> 4. board manager - is this the same as the chairperson of a board of directors?  If so, we can add that one ***, which would be the same as chairman, chairperson, company chairman, company director - we do have board member, which you could map to, and we have inside/outside board member as well if you know which applies
> 5. is a board of advisors distinct from a board of directors?  If so, we will have to think about whether or not it is within scope
> 6. COO - we could add them as needed, but we do have corporate officer you could map to for now
> 7. Chartered accountant - we have auditor, but that's different from an accountant for a firm, and the question would be do we need the accounting firm and/or the individual person that is the responsible party / primary partner from an accountancy that is responsible for signing off on the work that the outside accounting team performed?
> 8. chief creative officer - map to corporate officer
> 9. company formation agent - this is the same as an agent for service of process, which we do have
> 10. company secretary - this is in BE executives
> 11. director - if they are a board member then we would map to that.  if they are an employee with the title of director then no
> 12. executive chairman - this is the same as chief executive officer
> 13. executive director - this depends - we likely don't have this, which would be a corporate officer for a not for profit, but you could use corporate officer
> 14. financial advisor might be in Dev - we should think about adding this ...
> 15. investment banker - we have investment bank, but not the role for the person, but you could relate them to the organization using organization member
> 16. initial board of directors - ?? we have board of directors
> 17. we have Investor
> 18. managing director - this is on my list ***
> 19. mortgage advisor - if this is not in loans we should add it ***
> 20. non-executive director -- this is the same as non-executive board member
> 21. president - we should add this ***
> 22. private equity investor -- maps to investor
> 23. secretary -- we have this if it is company/corporate secretary
> 24. sole trader - this is there, maps to sole proprietor
> 25. top executive -- maps to chief executive officer
> 26. treasurer -- might be the same as chief financial officer ?  we have that
> 
> So - there are a few gaps I would add, but a number of the others are out of scope for FIBO I think, including the ones marked ***
> I'm also considering adding an employment situation, with employer and employee roles. Some of the other roles could be mapped to employee ... most banks have a gazillion vice presidents ...

Elisa, thank you for the detailed analysis :-) In some cases, you wrote, "... this is the same as ..." (e.g. non-executive director -- this is the same as non-executive board member). So, shouldn't we add synonyms?

**Comment by trypuz** [comment]
> Here’s my thoughts on this. 1. “role” vs “position” - The broader term is “role” such as the role of a department within an organization 2. Role has different enumerations based on its context * As given above, the role of a department has different values than the role of an employee 3. I wouldn’t use “position” in the context of a person’s role in an organization within FIBO since “position” in the investment industry refers to “Investor’s stake in a particular security or market”. (Barron’s) 4. Even the role of an employee in an organization depends on the type of organization. For example, for someone working at an exchange the roles would be trader, market-maker, specialist, etc. 5. Please don’t link role (or position) to title. For example, my title was Vice President of Product Management while my role was manager of the data model product. I’ve found that titles rarely specifies what the role actually is. * Also, please don’t even try to enumerate titles since that’s more of an HR responsibility and varies widely among organizations. A person given a title of VP in one organization will likely have way different roles and responsibilities from a VP in another organization. 6. Please don’t even try to enumerate roles as what Robert’s email seems to want to do. The problem is as I’ve been giving above, a role is highly dependent on the context.

John, I agree with your comments considering roles and positions. Considering your last point - I wish I had more details... In OpenCorporates we have just names of roles, no context is given.

**Comment by trypuz** [comment]
> The broader question is HOW should these be represented? To answer that @trypuz we need some use cases, which I know is something you champion: are there some for KYC?
> 
> Without any use cases, it would seem to me that none of these have any distinct properties or semantics, so they could each be represented as individuals of (new) class JobTitle ("Role" is overused), which would be referenced from individuals of class Position (see my response to #1236).

Pete, the use case is to map OpenCorporates structure data about officers to FIBO. The columns we can use for roles/positions mapping look as follows:
position | start_date | person_uid | end_date | occupation
director | 2006-10-05 | 11594785 | 2007-01-26 | FINANCIAL ADVISER

**Comment by rivettp** [comment]
@trypuz TBH mapping the data is not a use case - the key question is what needs to be done with the data once mapped? What are the competency questions? Is it a requirement to check that each Person assigned to a position has the required LegalCapacity? Which is something FIBO allows but I think is not at all useful. Similarly non-useful IMO is the FIBO ability to track who participated in electing a person to a Board position? These questions affect HOW the mapping is done. 
Other more pertinent questions:
- do we care that people are connected by working for the same company or specifically in the same unit (e.g. the Board)?
- do we care whether the position is as an employee or a (non-executive) director?
- do we care about general employees or only executives or the other specific job titles listed?
- do we care whether there is a temporal overlap (i.e. they were part of the company at the same time)?
  [note that AFAIK FIBO only tracks date of appointment not date of departure]
- in general do we care about history or only current data?
- is Open Corporates the ONLY source of interest or do we want something that can integrate data from other sources (e.g. SEC filings)?
- how are people identified - is there some sort of entity reconciliation? AFAIK Open Corporates has only simple names and I've seen the same person use multiple variations of their name e.g. shortened first names, different initials

**Comment by trypuz** [comment]
> @trypuz TBH mapping the data is not a use case - the key question is what needs to be done with the data once mapped? What are the competency questions? Is it a requirement to check that each Person assigned to a position has the required LegalCapacity? Which is something FIBO allows but I think is not at all useful. Similarly non-useful IMO is the FIBO ability to track who participated in electing a person to a Board position? These questions affect HOW the mapping is done.
> Other more pertinent questions:
> 
> * do we care that people are connected by working for the same company or specifically in the same unit (e.g. the Board)?
> * do we care whether the position is as an employee or a (non-executive) director?
> * do we care about general employees or only executives or the other specific job titles listed?
> * do we care whether there is a temporal overlap (i.e. they were part of the company at the same time)?
>   [note that AFAIK FIBO only tracks date of appointment not date of departure]
> * in general do we care about history or only current data?
> * is Open Corporates the ONLY source of interest or do we want something that can integrate data from other sources (e.g. SEC filings)?
> * how are people identified - is there some sort of entity reconciliation? AFAIK Open Corporates has only simple names and I've seen the same person use multiple variations of their name e.g. shortened first names, different initials

@rivettp, I've only pointed out at the technical aspect of the KYC use case related to the OpenCorporates data. IMO, your questions, though interesting by themselves and in general important for the KYC use case, are almost irrelevant for the mapping task I've described (we have in the OC data: person_uid, position and occupation, start_date and end_date, e.g. person with uid 11594785 has a position "director" and occupation "FINANCIAL ADVISER" from 2006-10-05 to 2007-01-26).
